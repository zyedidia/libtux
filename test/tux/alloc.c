#include <assert.h>
#include <stdlib.h>
#include <stdio.h>

int main() {
    // small allocations
    for (int i = 0; i < 10; i++) {
        size_t size = 1024;
        int* p = malloc(size);
        assert(p);
        printf("allocated: %ld\n", size);
    }

    // large allocations
    for (int i = 0; i < 10; i++) {
        size_t size = 1024 * 1024;
        volatile int* p = malloc(size);
        assert(p);
        for (int j = 0; j < 8192; j++) {
            p[j] = j;
        }
        printf("allocated: %ld\n", size);
    }
    return 0;
}
