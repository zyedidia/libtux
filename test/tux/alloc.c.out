allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1024
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
allocated: 1048576
